src/lib/client.ts

```diff
@@ -65,8 +65,19 @@ export class HTTPClient implements IRequestClient {
 
         axiosInstance.interceptors.response.use(
             (response) => response,
-            // eslint-disable-next-line @typescript-eslint/promise-function-async
-            (error) => Promise.reject(error.response.data),
+            // eslint-disable-next-line @typescript-eslint/promise-function-async,@typescript-eslint/tslint/config
+            (error) => {
+                if (!!error.response) {
```

Comment:
The double negation (`!!`) has the purpose of casting the element to be explicitly boolean. This is most useful in the case of not using type safe comparison operators. This may be the case in JS, but it is solved with TypeScript, which we use. This does not apply to our case. 

I also read more about the double negation and it seems that for conditions (like the case presented) it does not bring anything special, since JS has the concepts of 'truthy' and 'falsy'. 'falsy' means that the value of the variable is a default (e.g. false, 0, "", null), or undefined. For our case, this is all we need, and the double negation just reiterates that. 

I used the double negation to ensure that I have a boolean in the condition. However, diving more into what it means and if it brings anything better (code stability for instance) for this scenario, I chose to remove them. It keeps the code cleaner, closer to the examples in the [Axios docs](https://github.com/axios/axios#handling-errors), while having the same functionality.

Reply:
Sounds good. Please remove the double negation here, as it does not add anything to help the reader understand what is going on.

To be extra careful, we can also add `if (error.response && error.response.data)`